System.IO.Abstractions.TestingHelpers/MockFileStream.cs

```diff
@@ -65,6 +68,11 @@ public override void Flush()
 
         private void InternalFlush()
         {
+            if (disposed)
```

Comment:
I wouldn't evaluate this here. Maybe better to do it within `Close`/`Dispose`. This code is also executed when people call `Flush` directly, where I'd rather expect an exception when the stream is already closed.

Reply:
Moved to just the `Dispose` for the failing netcore branch anyway.